lantern/src/Cuda.cpp

```diff
@@ -35,3 +36,12 @@ void _lantern_cuda_show_config()
     std::cout << at::detail::getCUDAHooks().showConfig() << std::endl;
     LANTERN_FUNCTION_END_VOID
 }
+
+void* _lantern_cuda_get_device_capability(int64_t device)
+{
+    LANTERN_FUNCTION_START
+    cudaDeviceProp * devprop = at::cuda::getDeviceProperties(device);
```

Comment:
Simlarly you can wrap the body of the function in `#ifdef __CUDACC__` and just raise a `std::runtime_error()` if that is not defined.

Reply:
Okay, that is done too.

Reply:
Okay, __CUDACC__ does not seem to be defined even when compiling with ENV{CUDA} defined on my local computer (Windows). This: https://docs.nvidia.com/cuda/cuda-compiler-driver-nvcc/index.html#nvcc-identification-macro suggests __CUDACC__ is only defined when compiling `.cu` files, but __NVCC__ should be defined for `.cpp` files as well. Nevertheless testing for __NVCC__ does not seem to work either on my local computer. Some conversation here: https://github.com/boostorg/signals2/issues/53 suggests that nvcc by default uses VC++ for `.cpp` files anyway and so __NVCC__ might not be set in that case. I'm not sure if this is just a windows issue. I am going to make a commit to run CI, just to see whether this works on different platforms. We may have to set a custom define in the CMakeLists.txt based on `ENV{CUDA}` and test for that instead.

Reply:
Ohh ! It actually makes sense as it seems that `cuda_add_library` will only compile `.cu` files with nvcc.
It seems that we could add: `set_source_files_properties(test.cpp PROPERTIES LANGUAGE CUDA)` as suggested [here](https://stackoverflow.com/a/58861930/3297472) when CUDA is enabled - somewhere in here: https://github.com/mlverse/torch/blob/master/lantern/CMakeLists.txt#L133

Reply:
Okay, `set_source_files_properties(src/Cuda.cpp PROPERTIES LANGUAGE CUDA)` at first just caused `Cuda.cpp` to not get compiled at all. After adding `enable_language(CUDA)` to the `CMakeLists.txt`, `nvcc` was used to compile `Cuda.cpp` but then there was a compilation error. Likely something to do with incorrect flags or some such. Seemed like more trouble than it was worth, so I ended up just setting a definition using `set_source_files_properties(src/Cuda.cpp PROPERTIES COMPILE_DEFINITIONS __NVCC__)`. This solution works, at least on my machine. Is this an acceptable solution do you think? I could change the flag to something other than `__NVCC__` too, in case that makes it easier to remember it is something not being set by the compiler, but manually?

Reply:
This solution looks great @rdinnager ! Thanks! Let's leave `__NVCC__`, we can change it later if something comes up.